ttnn/cpp/ttnn/events.hpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <memory>
+
+#include "tt_metal/impl/device/device_mesh.hpp"
+
+namespace ttnn::events {
+
+struct MultiDeviceEvent
+{
+    MultiDeviceEvent(DeviceMesh* device_mesh);
+    std::vector<std::shared_ptr<Event>> events;
+};
+// Single Device APIs
+std::shared_ptr<Event> create_event(Device* device);
+void record_event(uint8_t cq_id, const std::shared_ptr<Event>& event);
+void wait_for_event(uint8_t cq_id, const std::shared_ptr<Event>& event);
+// Multi Device APIs
+MultiDeviceEvent create_event(DeviceMesh* device_mesh);
+void record_event(uint8_t cq_id, const MultiDeviceEvent& event);
+void wait_for_event(uint8_t cq_id, const MultiDeviceEvent& event);
+
+} // namespace ttnn::events
```

Comment:
does it make more sense to move this under `tt_metal/impl/event/event.hpp`

Reply:
Hey Joseph, the goal of adding these functions/classes in ttnn was to make them use multi-device and async mode which are currently ttnn concepts. I thought about placing these classes in the file you suggested, but was concerned about API leakages ... currently none of the objects in tt_metal interface with multiple devices or use async.

Reply:
after offline sync, this is fine